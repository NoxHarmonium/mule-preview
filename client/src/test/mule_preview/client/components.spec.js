import React from "react";
import renderer from "react-test-renderer";
import {
  MuleComponent,
  MuleContainer
} from "mule-preview/mule_preview.client.components";
import {
  jsToClj,
  makeAtom,
  makeSet
} from "mule-preview/mule_preview.client.test_utils";

describe("when rendering a Mule component", () => {
  describe("when the component has mapping", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <MuleComponent
            name="set-payload"
            description="Set some payload"
            css-class="some-css-class"
            content-root="some-content-root"
            location={jsToClj({ line: 1, position: 4 })}
            showing-atom={makeAtom(false)}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("when the component is unknown", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <MuleComponent
            name="sjdsfjlsdf"
            description="Set some payload"
            css-class="some-css-class"
            content-root="some-content-root"
            location={jsToClj({ line: 1, position: 4 })}
            showing-atom={makeAtom(false)}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

describe("when rendering a Mule container", () => {
  describe("when it has no children", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <MuleContainer
            name="flow"
            description="Do something"
            css-class="some-css-class"
            content-root="some-content-root"
            location={jsToClj({ line: 2, position: 3 })}
            showing-atom={makeAtom(false)}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("when it has children", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <MuleContainer
            name="flow"
            description="Do something"
            css-class="some-css-class"
            content-root="some-content-root"
            location={jsToClj({ line: 2, position: 3 })}
            showing-atom={makeAtom(false)}
          >
            <MuleComponent
              name="set-payload"
              description="Set some payload"
              css-class="some-css-class"
              content-root="some-content-root"
              location={jsToClj({ line: 1, position: 4 })}
              showing-atom={makeAtom(false)}
              labels={makeSet(["edited"])}
            />
            <MuleComponent
              name="message-properties-transformer"
              description="Transform something"
              css-class="some-css-class"
              content-root="some-content-root"
              location={jsToClj({ line: 4, position: 4 })}
              showing-atom={makeAtom(false)}
              labels={makeSet(["removed"])}
            />
            <MuleComponent
              name="http:request"
              description="Make some request"
              css-class="some-css-class"
              content-root="some-content-root"
              location={jsToClj({ line: 7, position: 4 })}
              showing-atom={makeAtom(false)}
              labels={makeSet(["added"])}
            />
          </MuleContainer>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
