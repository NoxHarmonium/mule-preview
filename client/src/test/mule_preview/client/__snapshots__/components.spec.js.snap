// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering a Mule component when the component has mapping renders correctly 1`] = `
<div
  className="rc-popover-anchor-wrapper display-inline-flex "
  style={
    Object {
      "WebkitFlex": "inherit",
      "flex": "inherit",
    }
  }
>
  <div
    className="rc-point-wrapper display-inline-flex"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitFlex": "auto",
        "WebkitFlexFlow": "column",
        "alignItems": "center",
        "flex": "auto",
        "flexFlow": "column",
      }
    }
  >
    <div
      className="component-container some-css-class"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="diff-icon"
      />
      <div
        className="component set-payload"
      >
        <img
          className="category-frame"
          src="/img/icons/org.mule.tooling.category.transformers.large.png"
        />
        <img
          className="icon"
          src="/img/icons/SetPayload-48x32.png"
        />
        <div
          className="label"
        >
          Set some payload
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`when rendering a Mule component when the component is unknown renders correctly 1`] = `
<div
  className="rc-popover-anchor-wrapper display-inline-flex "
  style={
    Object {
      "WebkitFlex": "inherit",
      "flex": "inherit",
    }
  }
>
  <div
    className="rc-point-wrapper display-inline-flex"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitFlex": "auto",
        "WebkitFlexFlow": "column",
        "alignItems": "center",
        "flex": "auto",
        "flexFlow": "column",
      }
    }
  >
    <div
      className="component-container some-css-class"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="diff-icon"
      />
      <div
        className="component sjdsfjlsdf"
      >
        <img
          className="category-frame"
          src="/img/icons/org.mule.tooling.ui.modules.core.miscellaneous.large.png"
        />
        <img
          className="icon"
          src="/img/icons/UnknownNode-48x32.png"
        />
        <div
          className="label"
        >
          Set some payload
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`when rendering a Mule container when it has children renders correctly 1`] = `
<div
  className="rc-popover-anchor-wrapper display-inline-flex "
  style={
    Object {
      "WebkitFlex": "inherit",
      "flex": "inherit",
    }
  }
>
  <div
    className="rc-point-wrapper display-inline-flex"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitFlex": "auto",
        "WebkitFlexFlow": "column",
        "alignItems": "center",
        "flex": "auto",
        "flexFlow": "column",
      }
    }
  >
    <div
      className="container mule-flow some-css-class"
    >
      <div
        className="container-title"
      >
        Do something
      </div>
      <div
        className="container-inner"
      >
        <div
          className="icon-container"
        >
          <img
            className="category-frame"
            src="/img/icons/org.mule.tooling.category.scopes.large.png"
          />
          <img
            className="icon container-image"
            src="/img/icons/sub-flow-48x32.png"
          />
        </div>
        <div
          className="container-children"
        >
          <div
            className="rc-popover-anchor-wrapper display-inline-flex "
            style={
              Object {
                "WebkitFlex": "inherit",
                "flex": "inherit",
              }
            }
          >
            <div
              className="rc-point-wrapper display-inline-flex"
              style={
                Object {
                  "WebkitAlignItems": "center",
                  "WebkitFlex": "auto",
                  "WebkitFlexFlow": "column",
                  "alignItems": "center",
                  "flex": "auto",
                  "flexFlow": "column",
                }
              }
            >
              <div
                className="component-container some-css-class"
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="diff-icon"
                />
                <div
                  className="component set-payload"
                >
                  <img
                    className="category-frame"
                    src="/img/icons/org.mule.tooling.category.transformers.large.png"
                  />
                  <img
                    className="icon"
                    src="/img/icons/SetPayload-48x32.png"
                  />
                  <div
                    className="label"
                  >
                    Set some payload
                  </div>
                </div>
              </div>
            </div>
          </div>
          <img
            className="flow-arrow"
            src="/img/arrow-right-2x.png"
          />
          <div
            className="rc-popover-anchor-wrapper display-inline-flex "
            style={
              Object {
                "WebkitFlex": "inherit",
                "flex": "inherit",
              }
            }
          >
            <div
              className="rc-point-wrapper display-inline-flex"
              style={
                Object {
                  "WebkitAlignItems": "center",
                  "WebkitFlex": "auto",
                  "WebkitFlexFlow": "column",
                  "alignItems": "center",
                  "flex": "auto",
                  "flexFlow": "column",
                }
              }
            >
              <div
                className="component-container some-css-class"
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="diff-icon"
                />
                <div
                  className="component message-properties-transformer"
                >
                  <img
                    className="category-frame"
                    src="/img/icons/org.mule.tooling.category.transformers.large.png"
                  />
                  <img
                    className="icon"
                    src="/img/icons/Transformer-48x32.png"
                  />
                  <div
                    className="label"
                  >
                    Transform something
                  </div>
                </div>
              </div>
            </div>
          </div>
          <img
            className="flow-arrow"
            src="/img/arrow-right-2x.png"
          />
          <div
            className="rc-popover-anchor-wrapper display-inline-flex "
            style={
              Object {
                "WebkitFlex": "inherit",
                "flex": "inherit",
              }
            }
          >
            <div
              className="rc-point-wrapper display-inline-flex"
              style={
                Object {
                  "WebkitAlignItems": "center",
                  "WebkitFlex": "auto",
                  "WebkitFlexFlow": "column",
                  "alignItems": "center",
                  "flex": "auto",
                  "flexFlow": "column",
                }
              }
            >
              <div
                className="component-container some-css-class"
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                <div
                  className="diff-icon"
                />
                <div
                  className="component http:request"
                >
                  <img
                    className="category-frame"
                    src="/img/icons/org.mule.tooling.category.cloudconnectors.large.png"
                  />
                  <img
                    className="icon"
                    src="/img/icons/http_connector_50x34.png"
                  />
                  <div
                    className="label"
                  >
                    Make some request
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`when rendering a Mule container when it has no children renders correctly 1`] = `
<div
  className="rc-popover-anchor-wrapper display-inline-flex "
  style={
    Object {
      "WebkitFlex": "inherit",
      "flex": "inherit",
    }
  }
>
  <div
    className="rc-point-wrapper display-inline-flex"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitFlex": "auto",
        "WebkitFlexFlow": "column",
        "alignItems": "center",
        "flex": "auto",
        "flexFlow": "column",
      }
    }
  >
    <div
      className="container mule-flow some-css-class"
    >
      <div
        className="container-title"
      >
        Do something
      </div>
      <div
        className="container-inner"
      >
        <div
          className="icon-container"
        >
          <img
            className="category-frame"
            src="/img/icons/org.mule.tooling.category.scopes.large.png"
          />
          <img
            className="icon container-image"
            src="/img/icons/sub-flow-48x32.png"
          />
        </div>
        <div
          className="container-children"
        />
      </div>
    </div>
  </div>
</div>
`;
